ros/src/twist_controller/twist_controller.py

```diff
@@ -4,11 +4,20 @@
 
 
 class Controller(object):
-    def __init__(self, *args, **kwargs):
-        # TODO: Implement
-        pass
-
-    def control(self, *args, **kwargs):
-        # TODO: Change the arg, kwarg list to suit your needs
-        # Return throttle, brake, steer
-        return 1., 0., 0.
+    def __init__(self, steer_ratio):
+        self.steer_ratio = steer_ratio
+
+    def control(self, target_linear_velocity, target_angular_velocity,
+                current_linear_velocity, dbw_status):
+        '''Defines target throttle, brake and steering values'''
+
+        # TODO: implement throttle controller
+        throttle = 0.5
+
+        # TODO: implement brake controller
+        brake = 0
+
+        # implement steering controller
+        steering = self.steer_ratio * target_angular_velocity
```

Comment:
Now you using `target_angular_velocity` to steering control, do you think we need CTE to smooth and safe drive?

Reply:
target_angular_velocity seems to be working for now.
CTE is a great idea but much harder to implement. Maybe we can improve it later making it smoother if needed?

Reply:
As for brake and throttle, we can have CTE through target velocity and current velocity.

Reply:
ok. I tried implementing CTE on my own([code](https://github.com/taikione/CarND-Capstone/blob/wp_updater_frenet/ros/src/waypoint_updater/waypoint_updater.py#L264)). I can send a pull request if necessary.

Reply:
Let's do the simple version (if someone confirms it works) and we can use your CTE when it is ready if it gives better results.

Reply:
OK, after finish learning the traffic light recognition model, I'll try to implement simple CTE code.